Sources/SwiftDocC/Infrastructure/External Data/Deprecated/OutOfProcessReferenceResolver+DeprecatedResolvers.swift

```diff
@@ -0,0 +1,126 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+private struct DeprecatedAPIError: DescribedError {
+    var apiName: String
+    init(apiName: String = #function) {
+        self.apiName = apiName
+    }
+    
+    var errorDescription: String {
+        return "\(apiName.singleQuoted) is deprecated and is no longer used."
+    }
+}
+
+// MARK: External Symbol Resolver
+
+@available(*, deprecated, renamed: "GlobalExternalSymbolResolver", message: "Use 'GlobalExternalSymbolResolver' instead. This deprecated API will be removed after 5.11 is released")
```

Comment:
It's about the scope of what's deprecated. 

If we didn't deprecate the individual `DocumentationContext` properties that held values of each protocol, then those would cause deprecation warning in `SwiftDocC` since they each reference a deprecated type. To silence those deprecation warnings and push them to the callers we need to deprecate those properties (since deprecate API is allowed to call other deprecated API). Since we aren't deprecated the entire documentation context we need to mark each property as deprecated individually.